test/puppetlabs/puppetdb/command_test.clj

```diff
@@ -482,21 +482,30 @@
                    (query-to-vec (format "SELECT certname, %s as catalog FROM catalogs"
                                          (sutils/sql-hash-as-str "hash")))))
             (is (= 0 (task-count delay-pool)))
-            (is (empty? (fs/list-dir (:path dlo)))))
+            (is (empty? (fs/list-dir (:path dlo))))))
 
-          (testing "should store the catalog if the node was deactivated after the message"
-            (with-message-handler {:keys [handle-message dlo delay-pool q]}
+        (testing "should store the catalog if the node was deactivated after the message"
+          (with-message-handler {:keys [handle-message dlo delay-pool q]}
 
-              (scf-store/delete-certname! certname)
-              (jdbc/insert! :certnames {:certname certname :deactivated tomorrow})
+            (scf-store/delete-certname! certname)
+            (jdbc/insert! :certnames {:certname certname :deactivated tomorrow})
 
-              (handle-message (queue/store-command q (make-cmd-req)))
+            (handle-message (queue/store-command q (make-cmd-req)))
 
-              (is (= [{:certname certname :deactivated tomorrow}]
-                     (query-to-vec "SELECT certname,deactivated FROM certnames")))
-              (is (= [{:certname certname :catalog catalog-hash}]
-                     (query-to-vec (format "SELECT certname, %s as catalog FROM catalogs"
-                                           (sutils/sql-hash-as-str "hash")))))
+            (is (= [{:certname certname :deactivated tomorrow}]
+                   (query-to-vec "SELECT certname,deactivated FROM certnames")))
+            (is (= [{:certname certname :catalog catalog-hash}]
+                   (query-to-vec (format "SELECT certname, %s as catalog FROM catalogs"
+                                         (sutils/sql-hash-as-str "hash")))))
+            (is (= 0 (task-count delay-pool)))
+            (is (empty? (fs/list-dir (:path dlo))))))
+
+        (testing "should store the proper certname for catalogs with underscores"
+          (let [certname "host_0"]
+            (with-message-handler {:keys [handle-message dlo delay-pool q]}
+              (handle-message (queue/store-command q (catalog->command-req version-num (assoc catalog :certname certname))))
```

Comment:
This is nested enough that I think `->>` might be a bit easier to follow?
```
(->> (assoc catalog :certname certname)
     (catalog->command-req version-num)
     (queue/store-command q)
     handle-message)
```
Possibly the same in the replace-facts test...